ClientTests/TabManagerNavDelegateTests.swift

```diff
@@ -0,0 +1,241 @@
+        sut.webViewWebContentProcessDidTerminate(anyWebView())
+
+        XCTAssertEqual(delegate1.receivedMessages, [.webViewWebContentProcessDidTerminate])
+        XCTAssertEqual(delegate2.receivedMessages, [.webViewWebContentProcessDidTerminate])
+    }
+
+    func test_webViewDidReceiveServerRedirectForProvisionalNavigation_sendsCorrectMessage() {
+        let sut = TabManagerNavDelegate()
+        let delegate1 = WKNavigationDelegateSpy()
+        let delegate2 = WKNavigationDelegateSpy()
+
+        sut.insert(delegate1)
+        sut.insert(delegate2)
+        sut.webView(anyWebView(), didReceiveServerRedirectForProvisionalNavigation: navigation)
+
+        XCTAssertEqual(delegate1.receivedMessages, [.webViewDidReceiveServerRedirectForProvisionalNavigation])
+        XCTAssertEqual(delegate2.receivedMessages, [.webViewDidReceiveServerRedirectForProvisionalNavigation])
+    }
+
+    func test_webViewDidStartProvisionalNavigation_sendsCorrectMessage() {
+        let sut = TabManagerNavDelegate()
+        let delegate1 = WKNavigationDelegateSpy()
+        let delegate2 = WKNavigationDelegateSpy()
+
+        sut.insert(delegate1)
+        sut.insert(delegate2)
+        sut.webView(anyWebView(), didStartProvisionalNavigation: navigation)
+
+        XCTAssertEqual(delegate1.receivedMessages, [.webViewDidStartProvisionalNavigation])
+        XCTAssertEqual(delegate2.receivedMessages, [.webViewDidStartProvisionalNavigation])
+    }
+
+    func test_webViewDecidePolicyFor_actionPolicy_sendsCorrectMessage() {
+        let sut = TabManagerNavDelegate()
+        let delegate1 = WKNavigationDelegateSpy()
+        let delegate2 = WKNavigationDelegateSpy()
+
+        sut.insert(delegate1)
+        sut.insert(delegate2)
+        sut.webView(anyWebView(),
+                    decidePolicyFor: WKNavigationAction(),
+                    decisionHandler: { _ in })
+
+        XCTAssertEqual(delegate1.receivedMessages, [.webViewDecidePolicyWithActionPolicy])
+        XCTAssertEqual(delegate2.receivedMessages, [.webViewDecidePolicyWithActionPolicy])
+    }
+
+    func test_webViewDecidePolicyFor_responsePolicy_sendsCorrectMessage() {
+        let sut = TabManagerNavDelegate()
+        let delegate1 = WKNavigationDelegateSpy()
+        let delegate2 = WKNavigationDelegateSpy()
+
+        sut.insert(delegate1)
+        sut.insert(delegate2)
+        sut.webView(anyWebView(),
+                    decidePolicyFor: WKNavigationResponse(),
+                    decisionHandler: { _ in })
+
+        XCTAssertEqual(delegate1.receivedMessages, [.webViewDecidePolicyWithResponsePolicy])
+        XCTAssertEqual(delegate2.receivedMessages, [.webViewDecidePolicyWithResponsePolicy])
+    }
+}
+
+// MARK: - Helpers
+
+private func anyWebView() -> WKWebView {
+    return WKWebView(frame: CGRect(width: 100, height: 100))
+}
+
+private func anyError() -> NSError {
+    return NSError(domain: "any error", code: 0)
+}
+
+private class WKNavigationDelegateSpy: WKNavigationDelegate {
```

Comment:
If we add conformance to NSObject we could remove extra code (from line 160 to 206), Leaving just the code related to the Spy class
`private class WKNavigationDelegateSpy: NSObject, WKNavigationDelegate {`

Reply:
Oh, you're right. Fixed it. Thank you!